### PR TITLE
Custom Table e2e Tests 3: Create and add writable DBs

### PR DESCRIPTION
**Pull request by iethree**
## Description
- as we add more tables, many of which we will want to test write actions against, we should make separate databases on the same DB servers for this purpose - so as not to dirty up datasets that we rely on remaining stable.
- Additions
  - adds a  `setupTestDB()` function that can create `testing_db` ephemeral databases in either our postgres or mysql containers.
  - adds an `is_writable` parameter to our `addQADatabase()` helper function to allow it to have conditional logic to use the correct credentials for writable DBs, and enables write actions in metabase on those DBs
  - uses the above to create writable snapshots for both postgres and mysql that we can use throughout our test suite.
  - adds a `queryTestDB()` function to easily run queries against the writable test DBs

Also adds tests that assert that the metabase API recognizes these DBs have actions enabled

Screen Shot 2023-01-31 at 3 02 06 PM

### CODE REVIEW

File: frontend/test/snapshot-creators/qa-db.cy.snap.js [hunk 1]
```diff
@@ -28,6 +29,20 @@ describe("qa databases snapshots", { tags: "@external" }, () => {
...
+    snapshot("mysql-writable");
+    deleteDatabase("mysqlID");
+
+    restoreAndAuthenticate();
+
+    setupTestDB("postgres");
+    addPostgresDatabase("Writable Postgres12", true);
+    snapshot("postgres-writable");
+    deleteDatabase("postgresID");
+
```
Comment (nemanjaglumac) [inline frontend/test/snapshot-creators/qa-db.cy.snap.js:45]: Do you maybe have an info handy re: how much time does this add to the snapshot generation?

File: frontend/test/__support__/e2e/cypress_data.js [hunk 2]
```diff
@@ -172,3 +172,24 @@ export const QA_DB_CONFIG = {
     },
   },
 };
+
+export const TESTING_DB_CONFIG = {
```
Comment (nemanjaglumac) [inline frontend/test/__support__/e2e/cypress_data.js:176]: Nit: WDYT about calling this `WRITABLE_DB_CONFIG`, that is calling that database "writable".

I had to double-check a few times in my head which one is QA and which one is testing DB.

File: frontend/test/metabase/scenarios/admin/databases/actions.cy.spec.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+describe(
+  "admin > database > external databases > enable actions",
+  { tags: "@external" },
+  () => {
+    ["mysql", "postgres"].forEach(dialect => {
+      it(`should show ${dialect} testing_db with actions enabled`, () => {
+        restore(`${dialect}-writable`);
+        cy.signInAsAdmin();
+
+        cy.request("/api/database/2").then(({ body }) => {
```
Comment (nemanjaglumac) [inline frontend/test/metabase/scenarios/admin/databases/actions.cy.spec.js:12]: Please be mindful of hard coded IDs.

File: frontend/test/snapshot-creators/qa-db.cy.snap.js (hunk 1 above)
Comment (nemanjaglumac) [reply frontend/test/snapshot-creators/qa-db.cy.snap.js:45]: FWIW, my first attempt to run Cypress on this branch failed on the snapshot generation step with "the sync too long".

It doesn't have to mean anything, but we should keep an eye on that.

**Review by nemanjaglumac — APPROVED** [review]
Left some comments that should be addressed, but overall this looks and works amazing!

File: frontend/test/snapshot-creators/qa-db.cy.snap.js (hunk 1 above)
Comment (iethree) [reply frontend/test/snapshot-creators/qa-db.cy.snap.js:45]: > Do you maybe have an info handy re: how much time does this add to the snapshot generation?

I just tested and it added 2 seconds on my machine: 13s -> 15s.

**Pull request merged**